### PR TITLE
Allows u8 as mouse binding value

### PR DESCRIPTION
**Pull request by Dav1dde**
Fixes: #2861

I did not remove "5" for compatibility reasons

### CODE REVIEW

**Comment by kchibisov** [comment]
>I did not remove "5" for compatibility reasons

Nah, I feel like we should remove this syntax, there's no need to keep bug around and there's no place for compatibility for such bugs before `1.0.0` in my opinion, since you'll throw a valid error.

**Comment by Dav1dde** [comment]
It's gone

File: CHANGELOG.md [hunk 1]
```diff
@@ -30,6 +30,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Tabstops not being reset with `reset`
 - Selection not cleared when switching between main and alt grid
 - Fallback to `LC_CTYPE=UTF-8` on macOS without valid system locale
+- Numbers are now allowed as mouse binding value
```
Comment (chrisduerr) [inline CHANGELOG.md:33]: If this is a new feature, it should be part of `Added`. If it's a fix for an old issue it should be properly documented.

I think I'd move it to changed section and go into more detail that mouse bindings are accepted again, but only as numbers, not as strings.

File: alacritty/src/config/bindings.rs [hunk 2]
```diff
@@ -621,7 +621,18 @@ impl<'a> Deserialize<'a> for MouseButtonWrapper {
             type Value = MouseButtonWrapper;
 
             fn expecting(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-                f.write_str("Left, Right, Middle, or a number")
+                f.write_str("Left, Right, Middle, or a number (0 <= x <= 255")
```
Comment (chrisduerr) [inline alacritty/src/config/bindings.rs:624]:
```suggestion
                f.write_str("Left, Right, Middle, or a number from 0 to 255")
```

File: alacritty/src/config/bindings.rs [hunk 3]
```diff
@@ -621,7 +621,18 @@ impl<'a> Deserialize<'a> for MouseButtonWrapper {
...
+
+            fn visit_u64<E>(self, value: u64) -> Result<MouseButtonWrapper, E>
+            where
+                E: de::Error,
+            {
+                if let Ok(other) = u8::try_from(value) {
+                    Ok(MouseButtonWrapper(MouseButton::Other(other)))
+                } else {
+                    Err(E::invalid_value(Unexpected::Unsigned(value), &self))
+                }
```
Comment (chrisduerr) [inline alacritty/src/config/bindings.rs:635]: Should probably use a match here.

**Comment by kchibisov** [comment]
@chrisduerr I guess it's already there https://github.com/alacritty/alacritty/blob/6b45780f3afa23e9e4277bf43e6bf381172fa1ea/alacritty.yml#L410

**Comment by chrisduerr** [comment]
> @chrisduerr I guess it's already there 

I just looked at the diff and you're completely right. My bad.

**Pull request merged**